### PR TITLE
Enable `EHCONT` and `CETCOMPAT` for some binaries on Windows

### PR DESCRIPTION
**Pull request by kouvel**
- Followup to https://github.com/dotnet/runtime/pull/55942
- Enabled `/guard:ehcont` as a linker option for EXEs and DLLs
- Enabled `/cetcompat` as a linker option for DLLs

### CODE REVIEW

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Comment by kouvel** [comment]
The changes to `configurecompiler.cmake` were a bit too global and affecting mono builds too. I have moved the addition of switches to more local places for coreclr. @janvorli could you please take another look?

File: eng/native/configurecet.cmake [hunk 1]
```diff
@@ -0,0 +1,9 @@
+# Enable EH-continuation table and CET-compatibility for native components for amd64 builds.
```
Comment (janvorli) [inline eng/native/configurecet.cmake:1]: It should still be included from the configurecompiler.cmake or rather moved back there (ideally co-located together like you have it here). All compiler options common for coreclr, libraries and installer should be there. We should just wrap it with an "if" that filters it out for Mono.

Comment (kouvel) [reply eng/native/configurecet.cmake]: Sounds good, done

**Review by janvorli — APPROVED** [review]
LGTM, thank you!

**Comment by jakobbotsch** [comment]
Does this require compiler updates? After this change I am getting 
```
  libucrtd.lib(cvt.obj) : error LNK2047: module contains C++ EH or complex EH metadata but was not compiled with /guard:ehcont
  libucrtd.lib(wcrtomb.obj) : error LNK2047: module contains C++ EH or complex EH metadata but was not compiled with /guard:ehcont
  libucrtd.lib(wcstombs.obj) : error LNK2047: module contains C++ EH or complex EH metadata but was not compiled with /guard:ehcont
  libucrtd.lib(debug_heap.obj) : warning LNK4291: module may contain '__except' (Structured Exception Handling) but was not compiled with /guard:ehcont; generating conservative metadata
  libucrtd.lib(getmodulefilenamea.obj) : error LNK2047: module contains C++ EH or complex EH metadata but was not compiled with /guard:ehcont
  libucrtd.lib(dbgrptt.obj) : warning LNK4291: module may contain '__except' (Structured Exception Handling) but was not compiled with /guard:ehcont; generating conservative metadata
  libucrtd.lib(terminate.obj) : warning LNK4291: module may contain '__except' (Structured Exception Handling) but was not compiled with /guard:ehcont; generating conservative metadata
  libucrtd.lib(argv_wildcards.obj) : error LNK2047: module contains C++ EH or complex EH metadata but was not compiled with /guard:ehcont
  libucrtd.lib(exit.obj) : warning LNK4291: module may contain '__except' (Structured Exception Handling) but was not compiled with /guard:ehcont; generating conservative metadata
  libucrtd.lib(onexit.obj) : error LNK2047: module contains C++ EH or complex EH metadata but was not compiled with /guard:ehcont
  libucrtd.lib(fdopen.obj) : warning LNK4291: module may contain '__except' (Structured Exception Handling) but was not compiled with /guard:ehcont; generating conservative metadata
  libucrtd.lib(fgetc.obj) : error LNK2046: module contains _local_unwind but was not compiled with /guard:ehcont
  libucrtd.lib(fputc.obj) : error LNK2046: module contains _local_unwind but was not compiled with /guard:ehcont
  libucrtd.lib(qsort.obj) : error LNK2047: module contains C++ EH or complex EH metadata but was not compiled with /guard:ehcont
  libucrtd.lib(fpctrl.obj) : warning LNK4291: module may contain '__except' (Structured Exception Handling) but was not compiled with /guard:ehcont; generating conservative metadata
  libucrtd.lib(ieee.obj) : warning LNK4291: module may contain '__except' (Structured Exception Handling) but was not compiled with /guard:ehcont; generating conservative metadata
  jit\clrjit_unix_x64_x64.dll : fatal error LNK1386: could not generate metadata for one or more modules not compiled with /guard:ehcont; use /force:guardehcont to link anyways
  ninja: build stopped: subcommand failed.
  BUILD: Error: native component build failed. Refer to the build log files for details.
      "D:\dev\dotnet\runtime\src\coreclr\..\..\artifacts\log\Debug\CoreCLR_windows__x64__Debug.log"
      "D:\dev\dotnet\runtime\src\coreclr\..\..\artifacts\log\Debug\CoreCLR_windows__x64__Debug.wrn"
      "D:\dev\dotnet\runtime\src\coreclr\..\..\artifacts\log\Debug\CoreCLR_windows__x64__Debug.err"
```

**Comment by kouvel** [comment]
The new switch `/guard:ehcont` requires compiler and linker updates, which I believe have been available for roughly a year or so. If static libraries are being used they would also have to be compiled with the new switch, mismatches would probably result in these kinds of errors.

**Comment by sandreenko** [comment]
> The new switch /guard:ehcont requires compiler and linker updates, which I believe have been available for roughly a year or so. If static libraries are being used they would also have to be compiled with the new switch, mismatches would probably result in these kinds of errors.

When I saw these errors I did git clean -dfx and updated VS to the last version (Version 16.11.0), and it did not help, what am I missing?

**Comment by kouvel** [comment]
That's pretty much all I did too, what errors are you seeing @sandreenko?

**Comment by sandreenko** [comment]
Thanks @jakobbotsch for the help, VS update does not install new SDKs automatically, so I needed to do "VS installer->modify->add Windows 10 SDK (10.0.19/20)"->install

**Pull request merged**